docs/README.md

```diff
@@ -0,0 +1,327 @@
+  and proposes changes to Swift (pre-1.0).
+  <!-- NOTE: Outdated -->
+
+### Archive
+
+These documents are known to be out-of-date and are superseded by other
+documentation, primarily [The Swift Programming Language][].
+They are preserved mostly for historical interest.
+
+- [AccessControl.rst](/docs/AccessControl.swift)
+- [Arrays.rst](/docs/Arrays.rst)
+  <!-- Has additional notes on bridging that may be of general interest? -->
+- [Generics.rst](/docs/Generics.rst)
+- [ErrorHandling.rst](/docs/ErrorHandling.rst)
+- [StringDesign.rst](/docs/StringDesign.rst)
+- [TextFormatting.rst](/docs/TextFormatting.rst)
+
+[The Swift Programming Language]: https://docs.swift.org/swift-book
+
+## External Resources
+
+The official [Swift blog](https://swift.org/blog/) contains a lot of useful information,
+such as how library evolution works
+and how the compiler's new diagnostic architecture is structured,
+helping us provide more precise diagnostics.
+
+TODO: Add a new document ExternalResources.md.
+
+## Uncategorized
+
+### Needs refactoring
+
+The documents in this section might be worth breaking up
+into several documents, and linking one document from the other.
+Breaking up into components will provide greater clarity
+to contributors wanting to add new documentation.
+
+- [ARCOptimization.rst](/docs/ARCOptimization.rst):
+  Covers how ARC optimization works, with several examples.
+  TODO: Not clear if this is intended to be a topic guide
+  or a reference guide.
+- [CompilerPerformance.md](/docs/CompilerPerformance.md):
+  Thoroughly discusses different ways of measuring compiler performance
+  and common pitfalls.
+  TODO: Consider breaking up into one high-level
+  topic guide explaining key concepts and individual how-to guides that
+  can be expanded independently.
+  Also, it's not the right place to explain details of different compiler modes.
+- [DevelopmentTips.md](/docs/DevelopmentTips.md):
+  Contains an assortment of tips for better productivity when working on the
+  compiler.
+  TODO: Might be worthwhile to make a dedicated how-to guide or topic guide.
+  It might also be valuable to introduce the tips in context, and have the
+  topic guide link to all the different tips.
+- [Diagnostics.md](/docs/Diagnostics.md):
+  Describes how to write diagnostic messages and associated educational notes.
+  TODO: Consider splitting into how-tos (e.g. for educational notes)
+  and recommended practices (on writing diagnostics).
+- [HowSwiftImportsCAPIs.md](/docs/HowSwiftImportsCAPIs.md):
+  Contains a thorough description of the mapping between
+  C/ObjC entities and Swift entities.
+  TODO: Not clear if this is intended to be language documentation
+  (for Swift developers), a topic guide or a reference guide.
+- [OptimizerCountersAnalysis.md](/docs/OptimizerCountersAnalysis.md):
+  TODO: Consider breaking up into a how-to guide
+  on dumping and analyzing the counters
```

Comment:
As with writing educational notes, I think that any time a topic is sufficiently specific like this, it probably won't fit into the bucket of "common tasks" (versus, say, "Testing"--which most certainly is common).

Reply:
I think I should remove the "common" adjective from "common tasks"; I think it's doing more harm than good. Instead I'll replace it with "specific tasks". Do you think that would alleviate this issue?

The "common" was intended as a -- there exists a subsystem X of the compiler/toolchain such that if you're working on subsystem X, you will have to do this at some point or another, and maybe frequently -- but I totally see how someone else would interpret it differently.

Reply:
Changed the word here "common" -> "specific".